--- /usr/share/devtools/makepkg-x86_64.conf	2019-09-22 00:10:53.000000000 +0800
+++ makepkg-aarch64.conf	2019-03-11 19:30:43.634352534 +0800
@@ -34,13 +32,15 @@
 # ARCHITECTURE, COMPILE FLAGS
 #########################################################################
 #
-CARCH="x86_64"
-CHOST="x86_64-pc-linux-gnu"
+CARCH="aarch64"
+CHOST="aarch64-unknown-linux-gnu"
 
 #-- Compiler and Linker Flags
+# -march (or -mcpu) builds exclusively for an architecture
+# -mtune optimizes for an architecture, but builds for whole processor family
 CPPFLAGS="-D_FORTIFY_SOURCE=2"
-CFLAGS="-march=x86-64 -mtune=generic -O2 -pipe -fno-plt"
-CXXFLAGS="-march=x86-64 -mtune=generic -O2 -pipe -fno-plt"
+CFLAGS="-march=armv8-a -O2 -pipe -fstack-protector-strong -fno-plt"
+CXXFLAGS="-march=armv8-a -O2 -pipe -fstack-protector-strong -fno-plt"
 LDFLAGS="-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now"
 #-- Make Flags: change this for DistCC/SMP systems
 #MAKEFLAGS="-j2"
