--- /usr/share/devtools/pacman-extra.conf	2019-09-22 00:10:53.000000000 +0800
+++ pacman-extra-armv6h.conf	2019-03-11 18:49:05.793639942 +0800
@@ -20,6 +20,6 @@
 #XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
 #CleanMethod = KeepInstalled
-Architecture = auto
+Architecture = armv6h
 
 # Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
 #IgnorePkg   =
@@ -74,16 +74,16 @@
 #Include = /etc/pacman.d/mirrorlist
 
 [core]
-Include = /etc/pacman.d/mirrorlist
+Include = /usr/share/devtools/qemu_mirrorlist
 
 [extra]
-Include = /etc/pacman.d/mirrorlist
+Include = /usr/share/devtools/qemu_mirrorlist
 
 #[community-testing]
 #Include = /etc/pacman.d/mirrorlist
 
 [community]
-Include = /etc/pacman.d/mirrorlist
+Include = /usr/share/devtools/qemu_mirrorlist
 
 # An example of a custom package repository.  See the pacman manpage for
 # tips on creating your own repositories.
