--- /usr/bin/archbuild	2019-06-11 00:11:13.000000000 +0800
+++ qemu_archbuild	2019-08-26 17:56:31.925524302 +0800
@@ -271,6 +271,11 @@
 	tag="${cmd%-build}"
 	repo=${tag%-*}
 	arch=${tag##*-}
+	if [[ "x${arch}" = "xaarch64" ]]; then
+		qemu_static='/usr/bin/qemu-aarch64-static'
+	else
+		qemu_static='/usr/bin/qemu-arm-static'
+	fi
 fi
 chroots='/var/lib/archbuild'
 clean_first=false
@@ -299,6 +304,8 @@
 # Pass all arguments after -- right to makepkg
 makechrootpkg_args+=("${@:$OPTIND}")
 
+. /usr/share/devtools/qemu_archbuild.sh
+
 if ${clean_first} || [[ ! -d "${chroots}/${repo}-${arch}" ]]; then
 	msg "Creating chroot for [%s] (%s)..." "${repo}" "${arch}"
 
@@ -316,15 +323,21 @@
 	rm -rf --one-file-system "${chroots}/${repo}-${arch}"
 	mkdir -p "${chroots}/${repo}-${arch}"
 	mkarchroot \
-		-C "/usr/share/devtools/pacman-${repo}.conf" \
+		-C "/usr/share/devtools/pacman-${repo}-${arch}.conf" \
 		-M "/usr/share/devtools/makepkg-${arch}.conf" \
+		-f "$qemu_static" \
+		-f '/usr/share/devtools/qemu_mirrorlist' \
+		-c "${pkgcache}/${arch}/" \
 		"${chroots}/${repo}-${arch}/root" \
 		"${base_packages[@]}" || abort
 else
 	lock 9 "${chroots}/${repo}-${arch}/root.lock" "Locking clean chroot"
 	arch-nspawn \
-		-C "/usr/share/devtools/pacman-${repo}.conf" \
+		-C "/usr/share/devtools/pacman-${repo}-${arch}.conf" \
 		-M "/usr/share/devtools/makepkg-${arch}.conf" \
+		-f "$qemu_static" \
+		-f '/usr/share/devtools/qemu_mirrorlist' \
+		-c "${pkgcache}/${arch}/" \
 		"${chroots}/${repo}-${arch}/root" \
                 pacman -Syu --noconfirm || abort
 fi
@@ -335,4 +348,4 @@
 fi
 
 msg "Building in chroot for [%s] (%s)..." "${repo}" "${arch}"
-exec makechrootpkg -r "${chroots}/${repo}-${arch}" "${makechrootpkg_args[@]}"
+exec makechrootpkg -r "${chroots}/${repo}-${arch}" -C "${pkgcache}/${arch}/" "${makechrootpkg_args[@]}"
