--- /usr/bin/archbuild	2019-01-30 00:11:39.000000000 +0800
+++ qemu_archbuild	2019-03-13 00:08:29.458016020 +0800
@@ -259,7 +259,7 @@
 }
 
 
-base_packages=(base-devel)
+base_packages=(base-devel namcap)
 makechrootpkg_args=(-c -n)
 
 cmd="${0##*/}"
@@ -299,6 +299,8 @@
 # Pass all arguments after -- right to makepkg
 makechrootpkg_args+=("${@:$OPTIND}")
 
+. /usr/share/devtools/qemu_archbuild.sh
+
 if ${clean_first} || [[ ! -d "${chroots}/${repo}-${arch}" ]]; then
 	msg "Creating chroot for [%s] (%s)..." "${repo}" "${arch}"
 
@@ -316,15 +318,21 @@
 	rm -rf --one-file-system "${chroots}/${repo}-${arch}"
 	mkdir -p "${chroots}/${repo}-${arch}"
 	mkarchroot \
-		-C "/usr/share/devtools/pacman-${repo}.conf" \
+		-C "/usr/share/devtools/pacman-${repo}-${arch}.conf" \
 		-M "/usr/share/devtools/makepkg-${arch}.conf" \
+		-f '/usr/bin/qemu-arm-static' \
+		-f '/usr/share/devtools/qemu_mirrorlist' \
+		-c "${pkgcache}/${arch}/" \
 		"${chroots}/${repo}-${arch}/root" \
 		"${base_packages[@]}" || abort
 else
 	lock 9 "${chroots}/${repo}-${arch}/root.lock" "Locking clean chroot"
 	arch-nspawn \
-		-C "/usr/share/devtools/pacman-${repo}.conf" \
+		-C "/usr/share/devtools/pacman-${repo}-${arch}.conf" \
 		-M "/usr/share/devtools/makepkg-${arch}.conf" \
+		-f '/usr/bin/qemu-arm-static' \
+		-f '/usr/share/devtools/qemu_mirrorlist' \
+		-c "${pkgcache}/${arch}/" \
 		"${chroots}/${repo}-${arch}/root" \
                 pacman -Syu --noconfirm || abort
 fi
@@ -335,4 +343,4 @@
 fi
 
 msg "Building in chroot for [%s] (%s)..." "${repo}" "${arch}"
-exec makechrootpkg -r "${chroots}/${repo}-${arch}" "${makechrootpkg_args[@]}"
+exec makechrootpkg -r "${chroots}/${repo}-${arch}" -C "${pkgcache}/${arch}/" "${makechrootpkg_args[@]}"
